core/src/replay_stage.rs

```diff
@@ -502,13 +501,15 @@ impl ReplayStage {
         banking_tracer: Arc<BankingTracer>,
         popular_pruned_forks_receiver: PopularPrunedForksReceiver,
     ) -> Result<Self, String> {
-        let mut tower = if let Some(process_blockstore) = maybe_process_blockstore {
-            let tower = process_blockstore.process_to_create_tower()?;
-            info!("Tower state: {:?}", tower);
-            tower
-        } else {
-            warn!("creating default tower....");
-            Tower::default()
+        let mut tower = match maybe_tower {
```

Comment:
might as well move this up to `validator.rs` and pass in a `mut Tower` instead of an option.

Reply:
Done.